doc/sphinx/source/fusion/pattern-matching.rst

```diff
@@ -0,0 +1,81 @@
+.. fusion/pattern-matching.rst: 
+
+Applying pattern matcher to fuse ops
+-------------------------------------
+
+Before delving into the details of pattern matching, note that the sole purpose 
+of the pattern-matching step is to **find** patterns in the framework's graph of 
+outputs.  A subgraph, then, could contain any operation nGraph's IR defines 
+(addition, subtraction, etc) along with some special wildcard nodes. 
+
+An analogy that is probably familiar to many programmers is that of the ``Regex``, 
+AKA "Regular expression". Just as one can write a regex (pattern) and run it 
+through some text to find and/or replace the occurences of that pattern in 
+the given text, so too can one write optimization-passes to construct patterns 
+which are just regular nGraph graphs and run those patterns through given graphs.
+
+In the ``Regex`` analogy, 
+
+
+* Letters (for example: `A`, `B`, `m`, `n`), 
+
+* Strings (for example: `ABBA`, `Hello world`), and
+
+* Collective Symbols (for some regex programs they are `*`, `.`)
+
+
+respectfully correspond to: 
+
+.. Letter -> Node
+
+* ``ngraph::Node`` (``op::Add``, ``op::BatchNormBackprop``), 
+
+.. Strings
+
+* Graphs consisting of ``ngraph::Nodes``, and
+
+.. Collective Symbols 
+
+* ``op::*`` (for some graph programs they are ``pattern::op::Label``, ``pattern::op::Skip``)
+
+where Operators need arguments, and Leaves cannot take arguments.  
+
+
+At the lower level, the nGraph C++ API is:  
+
+.. doxygenclass:: ngraph::pattern::Matcher
+   :project: ngraph
+   :members:
+
+
+To create a trivial graph representing ``-(-A) = A``:
+
+.. code-block:: cpp 
+
+   auto a = make_shared(element::i32, shape); 
```

Comment:
This should be making a `Parameter`. The code block should be the same as the one below that makes the pattern. But I think the pattern would not actually match since the parameters would not be the same. @Krovatkin ?